### PR TITLE
Use 'assert' for runtime assertions during development

### PR DESCRIPTION
**Pull request by markdalgleish**
See changeset for details.

Note: We'll merge this once sku v10.3.0 has been out for a while, but I'm opening the PR now so the code doesn't rot in the meantime.

### CODE REVIEW

**Comment by maraisr** [comment]
Curious; why didn't you use the `invariant`/`warning` methods typically used in the react space? The hyped `babel-plugin-dev-expression` already culls those things during builds, and optionally produce error codes - that map back to the English text.

**Comment by markdalgleish** [comment]
Because it's decoupled from `NODE_ENV`. We want to be able to build Playroom in production mode but retain the assertions.

**Comment by maraisr** [comment]
Sucks that second-grade is needed to be used; when tooling couple themselves to NODE_ENV. Imo that plugin should have a similar development mode that the react plugin has - so you can simply turn it of/on.

File: lib/components/MenuItem/MenuItem.tsx [hunk 1]
```diff
@@ -39,14 +40,13 @@ export const MenuItem = ({ children, onClick, data }: MenuItemProps) => {
...
-    } else {
-      throw new Error('MenuItem rendered outside menu context');
-    }
+  assert(
+    menuContext !== null,
+    'A MenuItem must be rendered as an immediate child of a Menu. See the documentation for correct usage: https://seek-oss.github.io/braid-design-system/components/MenuRenderer',
+  );
+
+  if (menuContext === null) {
+    throw new Error();
```
Comment (michaeltaranto) [inline lib/components/MenuItem/MenuItem.tsx:49]:
```suggestion
    throw new Error('MenuItem rendered outside menu context');
```

**Pull request merged**